### ISSUE TITLE
Add a setting to always ask to save untitled workspaces 

### ISSUE THREAD

**Issue by sentry07** [microsoft/vscode#69943]
Issue Type: <b>Bug</b>

Prereq: HotExit disabled; Restore Windows disabled

1) Open folder in VSCode
2) Add folder to workspace (creating a dirty workspace)
3) Close VSCode window (File->Exit, X in top right, doesn't matter)
4) Open different folder in VSCode

At this point the new folder opens in one window, and the dirty workspace opens in a second window. The order in which you close the windows changes how the dirty workspace is handled.

1) If you close the dirty workspace window FIRST, it will ask you if you'd like to save your workspace settings as a file. Yes or no, this will stop the dirty workspace from being reopened unintentionally.
2) If you close the dirty workspace window LAST, it will ask you nothing and the cycle repeats.

With HotExit disabled, closing a window with a dirty workspace should just discard the workspace, correct? Or does HotExit only apply to files themselves and not the workspace? If this is the case, then it's a feature request and not a bug. :)

VS Code version: Code - Insiders 1.32.0-insider (507312a3e3b34b084b467dfd983263bc7c9d87e0, 2019-03-06T10:06:48.833Z)
OS version: Windows_NT x64 6.1.7601

</details>Extensions: none

**Comment by bpasero** [comment]
@sentry07 yeah, an untitled workspace is actually totally independent from the hot-exit feature, they have no impact on each other. So your feature request would be to discard untitled workspaces upon exit?

**Comment by sentry07** [comment]
VSCode asks you if you want to save the workspace if you explicitly close the workspace before closing the window, or if the window is NOT the last window to be closed. I propose this should be the default action when closing a window with an untitled workspace.

**Comment by bpasero** [comment]
@sentry07 to be clear, you want to always be asked if you want to save the workspace even when you do File > Exit right? We chose against this option so that having untitled workspaces does not ask you each time when you quit.

**Comment by sentry07** [comment]
I feel like saving the untitled workspace and reopening it any time you reopen VSCode is solving a problem that didn't need to be solved. If the person wanted to save the workspace, having VSCode ask if they want to save the workspace when closing the window would serve the purpose. If they didn't want to save it, then they would say no to the dialog box. If you're going to have a HotExit function so people can close the window without saving their work, and you allow it to be disabled, why wouldn't it disable all the HotExit functions including workspaces? Or why not create a Workspace HotExit setting so that can be disabled separately? I'd be completely ok with that.

I'm not trying to be a dick in asking this. I just started using VSCode a couple weeks ago. I wrote an extension for the programming language I use that can add a global include folder to the workspace and it changed the behavior of VSCode because of the untitled workspace functionality being different. I'm just trying to figure out why workspaces behave differently than files and what to do to fix it.
